src/librustc_typeck/check/wfcheck.rs

```diff
@@ -26,7 +26,7 @@ use errors::{DiagnosticBuilder, DiagnosticId};
 use rustc::hir::intravisit::{self, Visitor, NestedVisitorMap};
 use rustc::hir;
 
-pub struct CheckTypeWellFormedVisitor<'a, 'tcx:'a> {
+pub struct CheckTypeWellFormed<'a, 'tcx:'a> {
```

Comment:
It seemed best to me to split the code that visits the HIR from the code which actually does the wf calculations. At this point, this struct probably isn't that useful and its methods could just be free functions. What do you think?

Reply:
Yeah, I think it would make sense to transform them into free-standing functions. That makes it easier to see that there's no state in `self` that one needs to worry about.